lib/posthog/client.rb

```diff
@@ -399,12 +400,10 @@ def add_local_person_and_group_properties(distinct_id, groups, person_properties
       all_person_properties = { distinct_id: distinct_id }.merge(person_properties)
 
       all_group_properties = {}
-      if groups
-        groups.each do |group_name, group_key|
-          all_group_properties[group_name] = {
-            :'$group_key' => group_key
-          }.merge((group_properties && group_properties[group_name]) || {})
-        end
+      groups&.each do |group_name, group_key|
+        all_group_properties[group_name] = {
+          '$group_key': group_key
+        }.merge((group_properties && group_properties[group_name]) || {})
```

Comment:
I think this is valid?

```suggestion
        }.merge((group_properties&.[group_name]) || {})
```

Reply:
ah, no, things were unhappy with that

Reply:
It does exist, but it's probably 3.3+ :) ty for trying